### PR TITLE
i18n tooling, fixes, more languages plus Transifex hosting

### PR DESCRIPTION
**Pull request by OmarIthawi**
We'd like to add more languages such as Canadian French as a supported language. This is a continuation to https://github.com/Stanford-Online/xblock-free-text-response/pull/35 , I should have completed the work then, but didn't.

Before you continue, take a look at [edx/i18n-tools](https://github.com/edx/i18n-tools) and [edx/cookiecutter-django-app](https://github.com/edx/cookiecutter-django-app).

### Summary of the Changes
 - Switched from `django.utils.translation.ugettext` to `i18n_service.ugettext` to capture what's in the `text.po` files.
 - Added a Makefile to help manage the translations. The file is copied and modified from: [edx/cookiecutter-django-app](https://github.com/edx/cookiecutter-django-app)
 - Added `freetextresponse/locale/config.yaml` which works as the configuration file of the `i18n_tools` command 
 - Refactored the linking of `locale` link to `translations` to streamline it with the current (but undocumented) conventions. See https://github.com/open-craft/xblock-poll/pull/43 for an example of adding full i18n support to an XBlock.
 - Added `requirements-dev.txt` to pin development packages. Used for now only for translations.

### TODO

 - [x] Make Travis happy
 - [x] Push to the [Open edX XBlocks project on Transifex](https://www.transifex.com/open-edx/xblocks/content/)
 - [x] Document how to update the strings properly from Transifex
 - [x] Manually test that the the Arabic translation is working
 - [ ] I would like to pin `pylint`  and others in `requirements-dev.txt`. Unversioned style checker is a mess to follow up with.

### It Works!

<kbd>image</kbd>

### CODE REVIEW

**Comment by OmarIthawi** [comment]
Hello @stvstnfrd! Thanks for passing by. I'm almost done with this. However, I'm a bit surprised by the fact that it's not possible to pull `vi` language from Transifex!

How important is it to keep this language?

**Comment by OmarIthawi** [comment]
@stvstnfrd good news. The marathon is over, I've been able to fix all the known issues and now it works with Arabic, Vietnamese and Canadian French.

Please take a look and let me know if you have any notes.

**Comment by OmarIthawi** [comment]
@ahmedaljazzar and @melvinsoft, could you please take a quick look here?

**Comment by OmarIthawi** [comment]
@stvstnfrd @caesar2164 A gentle reminder about this open pull request.

**Comment by iamjazzar** [comment]
@OmarIthawi I'm happy with the changes so far. I'll test it locally and I'll let you know how it goes.

**Comment by OmarIthawi** [comment]
@stvstnfrd @caesar2164 could you please check this out? I'm guessing it's the Stanford-wide university time off.

**Comment by OmarIthawi** [comment]
@stvstnfrd @caesar2164 pretty please :pray:

**Review by caesar2164 — COMMENTED** [review]
Hi @OmarIthawi, sorry for not replying sooner. We are actually in the middle of modernizing all our xblocks, so your reviewing your PR got delayed.

Could you actually look at this PR: https://github.com/Stanford-Online/xblock-free-text-response/pull/43

We are hoping to make your changes as part of that larger set of work.

File: setup.py [hunk 1]
```diff
@@ -57,6 +57,7 @@ def run_tests(self):
         "freetextresponse": [
             'public/*',
             'templates/*',
+            'translations/*',
```
Comment (stvstnfrd) [inline setup.py:60]: Change exists in https://github.com/Stanford-Online/xblock-free-text-response/pull/43 .

File: setup.py [hunk 2]
```diff
@@ -47,7 +47,7 @@ def run_tests(self):
     ],
     entry_points={
         'xblock.v1': [
-            'freetextresponse = freetextresponse:FreeTextResponse',
+            'freetextresponse = freetextresponse.freetextresponse:FreeTextResponse',
```
Comment (stvstnfrd) [inline setup.py:50]: Similar change exists in https://github.com/Stanford-Online/xblock-free-text-response/pull/43 .

File: .travis.yml [hunk 3]
```diff
@@ -20,5 +20,6 @@ branches:
 env:
     - TOXENV=py27-dj18
     - TOXENV=coveralls
-    - TOXENV=pep8
+    - TOXENV=pycodestyle
```
Comment (stvstnfrd) [inline .travis.yml:23]: Implemented on https://github.com/Stanford-Online/xblock-free-text-response/pull/43

File: tox.ini [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
 [testenv]
 commands = {envpython} manage.py test
 
-[testenv:pep8]
+[testenv:pycodestyle]
 deps =
-    pep8
-commands = {envbindir}/pep8 freetextresponse/
+    pycodestyle
+commands = {envbindir}/pycodestyle freetextresponse/
```
Comment (stvstnfrd) [inline tox.ini:11]: Implemented on https://github.com/Stanford-Online/xblock-free-text-response/pull/43

File: package.json [hunk 5]
```diff
@@ -1,22 +0,0 @@
...
-    "url": "https://github.com/Stanford-Online/xblock-free-text-response.git"
-  },
-  "bugs": {
-    "url": "https://github.com/Stanford-Online/xblock-free-text-response/issues"
-  },
-  "keywords": [
-    "openedx",
-    "xblock"
-  ]
-}
```
Comment (stvstnfrd) [inline package.json:22]: On https://github.com/Stanford-Online/xblock-free-text-response/pull/43,
we still use a lightweight `package.json` to install node-based dependencies:
- `csslint`
- `eslint`
- `less`

File: freetextresponse/translations/ar/LC_MESSAGES/text.po [hunk 6]
```diff
@@ -0,0 +1,231 @@
+# SOME DESCRIPTIVE TITLE.
```
Comment (stvstnfrd) [inline freetextresponse/translations/ar/LC_MESSAGES/text.po:1]: I pulled in the new strings you had translated:
https://github.com/Stanford-Online/xblock-free-text-response/pull/43

File: conf/locale [hunk 7]
```diff
@@ -1 +0,0 @@
-../freetextresponse/locale/
```
Comment (stvstnfrd) [inline conf/locale:1]: Removed: https://github.com/Stanford-Online/xblock-free-text-response/pull/43

File: freetextresponse/__init__.py [hunk 8]
```diff
@@ -3,4 +3,3 @@
 Instructors can specify a list of phrases, of which one must be
 present in order for the student to receive credit.
 """
-from .freetextresponse import FreeTextResponse
```
Comment (stvstnfrd) [inline freetextresponse/__init__.py:6]: Removed: https://github.com/Stanford-Online/xblock-free-text-response/pull/43

**Review by stvstnfrd — COMMENTED** [review]
I think I have represented the significant changes here in the related PR:
https://github.com/Stanford-Online/xblock-free-text-response/pull/43

**Comment by stvstnfrd** [comment]
@OmarIthawi Thanks! I think I have everything covered in https://github.com/Stanford-Online/xblock-free-text-response/pull/43 , so I'm going to close this in favor of it.

**Comment by OmarIthawi** [comment]
Thanks @stvstnfrd for making this XBlock and improving it!

**Pull request closed without merging**